jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/TypescriptCodeWhispererFileCrawler.kt

```diff
@@ -9,15 +9,14 @@ import com.intellij.psi.PsiFile
 object TypescriptCodeWhispererFileCrawler : CodeWhispererFileCrawler() {
     override val fileExtension: String = "ts"
     override val dialects: Set<String> = setOf("ts", "tsx")
-
-    override val testFilenamePattern: Regex = """^.*\.test\.(ts|tsx)${'$'}""".toRegex()
-
-    // TODO: Add implementation when UTG is enabled
-    override fun guessSourceFileName(tstFileName: String): String = ""
+    override val testFileNamingPatterns: List<Regex> = listOf(
+        Regex("""^(.+)\.test(\.ts|\.tsx)$"""),
+        Regex("""^(.+)\.spec(\.ts|\.tsx)$""")
```

Comment:
If it's for nameing conventions, should we also enable `Test` with upper case `T`? I have also seen that usage.

Reply:
i see, so we might have like `aJavaScript.Test.ts`

Reply:
let me add it

Reply:
updated